src/hooks/useFormReducer.ts

```
@@ -0,0 +1,178 @@
+  | { type: "CHANGE"; payload: { key: keyof T; value: T[keyof T] } }
+  | { type: "CHANGE_MULTIPLE"; payload: Partial<T> }
+  | { type: "RESET" }
+  | { type: "RESET_FIELD"; payload: keyof T };
+
+interface UseFormReducerOptions<T> {
+  validators?: Partial<Record<keyof T, (value: T[keyof T]) => boolean>>;
+  formatters?: Partial<Record<keyof T, (value: string) => string>>;
+  onChange?: (state: T) => void;
+}
+
+export function useFormReducer<T extends Record<string, any>>(
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)